var/spack/repos/builtin/packages/adios/package.py

```diff
@@ -12,30 +36,107 @@ class Adios(Package):
     """
 
     homepage = "http://www.olcf.ornl.gov/center-projects/adios/"
-    url = "https://github.com/ornladios/ADIOS/archive/v1.9.0.tar.gz"
+    url      = "https://github.com/ornladios/ADIOS/archive/v1.10.0.tar.gz"
+
+    # make -j is currently not supported
+    parallel = False
+
+    version('1.10.0', 'eff450a4c0130479417cfd63186957f3')
+    version('1.9.0' , '310ff02388bbaa2b1c1710ee970b5678')
+    version('1.8.0' , 'c9116ec31e6b386f0e6ea5ab675c57e9')
+    version('1.7.0' , '266897ee3a390985d840bbe2935b4293')
+    version('1.6.0' , 'b3c735f3afbf552ec0d5e8152bffd778')
+
+    variant('shared', default=True,
+            description='Builds a shared version of the library')
 
-    version('1.9.0', '310ff02388bbaa2b1c1710ee970b5678')
+    variant('fortran', default=True,
+            description='Enable Fortran bindings support')
+
+    variant('mpi', default=False, description='Enable MPI support')
+    variant('infiniband', default=False, description='Enable infiniband support')
+
+    variant('zlib', default=True, description='Enable szip transform support')
+    variant('szip', default=False, description='Enable szip transform support')
+    variant('hdf5', default=False, description='Enable HDF5 transport support')
+    variant('netcdf', default=False, description='Enable NetCDF transport support')
 
     # Lots of setting up here for this package
     # module swap PrgEnv-intel PrgEnv-$COMP
     # module load cray-netcdf/4.3.3.1
     # module load cray-hdf5/1.8.14
     # module load python/2.7.10
-    depends_on('hdf5')
-    depends_on('mxml')
+
+    depends_on('autoconf')
```

Comment:
if i understand correctly, `mxml@2.8` should not be used with ADIOS? Then `depends_on('mxml@2.9:`)` would be the correct way.

My comment above about linking was related to a hypothetical problem which can appear in general. Perhaps it is very unlikely in this case, but for the sake of consistency I think it is a good idea to list all dependencies as `depends_on`, even those which can be provided by a package. 

If you feel like there could be any advantage in using internal `mxml` in `adios@1.10:` as opposed to the external `mxml@2.9`, then one could add a variant `internal-mxml` with `default=False`.

p.s. i know nothing about `adios`, i comment from a general perspective to minimize possible issues and hopefully improve the quality of the package.